src/test/operators/join_multi_predicates_test.cpp

```diff
@@ -0,0 +1,368 @@
+                       2,
+                       {{_column_pair_2, PredicateCondition::GreaterThan}}};
+  }
+
+ protected:
+  void SetUp() override { JoinTest::SetUp(); }
+
+  void _test_join_output(const JoinParameters params) {
+    test_join_output<JoinType>(params.table_pair.first, params.table_pair.second, params.first_predicate.column_ids,
+                               params.first_predicate.predicate_condition, params.join_mode,
+                               params.expected_result_table_file_path, params.chunk_size, params.additional_predicates);
+  }
+
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_a;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_a_nulls_first;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_a_nulls_last;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_a_nulls_random;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_a2_nulls_random;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_b_larger;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_b_nulls_first_larger;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_b_nulls_last_larger;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_b_nulls_random;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_b_nulls_random_larger;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper_b2_nulls_random_larger;
+
+  inline static ColumnIDPair _column_pair_1{ColumnID{0}, ColumnID{2}};
+  inline static ColumnIDPair _column_pair_2{ColumnID{1}, ColumnID{1}};
+  inline static ColumnIDPair _column_pair_3{ColumnID{2}, ColumnID{0}};
+
+  inline static std::optional<JoinParameters> _base_choice_join_parameters;
+};
+
+// Input Domain Modeling
+
+// multi predicate join characteristics
+// ---------------
+// [A] join mode
+//    [1] Inner
+//    [2] Left
+//    [3] Right
+//    [4] Full outer
+//    [5] Cross
+//    [6] Semi
+//    [7] AntiDiscardNulls
+// [B] predicate condition used at least once
+//    [1] GreaterThan
+//    [2] GreaterThanEquals
+//    [3] LessThanEquals
+//    [4] LessThan
+//    [5] Equals
+//    [6] NotEquals
+// [C] null values in value comparisons
+//    [1] left
+//    [2] right
+//    [3] both
+//    [4] none
+// [D] condition satisfaction
+//    [1] none
+//    [2] 1. predicate
+//    [3] 1. & 2. predicate
+//    [4] all predicates
+// [E] null value distribution in columns
+//    [1] random
+//    [2] nulls last
+//    [3] nulls first
+// [F] size of join columns
+//    [1] left > right
+//    [2] left < right
+//    [3] same sizes
+// [G] number of additional predicates
+//    [1] 1
+//    [2] >1
+// [H] data type relation between join columns
+//    [1] equal
+//    [2] not equal
+// [I] column data types
+//    [1] int
+//    [2] string
+
+// Choosing combinations using the Base Choice strategy:
+
+// Base Choice:
+// A1 B5 C2 D3 E1 F2 G1 H1 I1       InnerLTableSmallerRTableRandomNullsEqualsGreaterThan
```

Comment:
Thanks for not attempting to test all possible combinations!!

Reply:
And thanks for actually testing on table input and table output! This is great for verifying tests!